### PR TITLE
Rfc/client filter pi

### PR DESCRIPTION
**Pull request by gdchamal**
Add a proposal RFC for client PI filtering principle using X-Caliopen-PI header

Corrections also on current message api rfc, lot or rewording and remove of remarks

### CODE REVIEW

File: _posts/rfc/2016-04-01-PI-filter.md [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+## Principle
+
+An HTTP Header **X-Caliopen-PI** is used to give PI range wanted by an API Client.
+Format for this header is an integer range between 0 and 100, inclusives values.
+Values are separated using semi colon ';' character.
+
+### Examples:
+
+X-Caliopen-PI: 0;100 -> to retrieve all resources
```
Comment (iamdey) [inline _posts/rfc/2016-04-01-PI-filter.md]: what if resource has an undefined PI ?

* always omitted ?
* given if min value is 0 ?

Comment (iamdey) [inline _posts/rfc/2016-04-01-PI-filter.md]: what if max is lower than min: `X-Caliopen-PI: 80;60`

File: _posts/rfc/2016-04-01-PI-filter.md [hunk 2]
```diff
@@ -0,0 +1,31 @@
...
+
+X-Caliopen-PI: 50;100 -> to retrieve only resources having at least a PI value of 50
+
+## Return code
+
+API will reply with a 403 HTTP status code if a wanted resource PI do not fit into
```
Comment (iamdey) [inline _posts/rfc/2016-04-01-PI-filter.md]: what about : 

```
417 Expectation Failed
    The server cannot meet the requirements of the Expect request-header field.
```

File: _posts/rfc/2015-06-10-message-api.md [hunk 3]
```diff
@@ -20,7 +20,18 @@ mockups.
...
+Message related resources are **Recipient**, **Thread**, **Attachment** and **Message**.
+
+## Recipient
+
+A recipient MIGHT be related to an known **Contact**. It is defined by
+the following informations:
+
+* `contact_id`: Known user contact reference
+* `address`: Address to use for sending a message to this recipient
+* `protocol`: Communication protocol for this recipient address
```
Comment (iamdey) [inline _posts/rfc/2015-06-10-message-api.md]: the sender information is missing

File: _posts/rfc/2015-06-10-message-api.md [hunk 4]
```diff
@@ -29,37 +40,42 @@ Every **Thread** MUST provide the following information:
...
 A **Message** can come from several providers (email, twitter, instant chat...).
 This is a common abstraction, every message MUST provide the following
 information:
 
-* `status`: read, draft, unread... **<-- Pleas provide some insight on the way
-  you think it should be handled**
+* `status`: draft, sending, sent, deleted for outcoming messages. unread, read, deleted 
+  for incoming ones.
 * `thread`: link to the thread the message is included in.
-* `sender`: link to contact who sent this message if any. **<--- should raw
```
Comment (iamdey) [inline _posts/rfc/2015-06-10-message-api.md]: As explain, we need to know who has send the message

File: _posts/rfc/2016-04-01-PI-filter.md (hunk 1 above)
Comment (gdchamal) [inline _posts/rfc/2016-04-01-PI-filter.md]: no PI = 0 then not ommited only if min = 0

HTTP BadRequest (400) raised if min > max and if min < 0 and max > 100. I update return code section with that

File: _posts/rfc/2016-04-01-PI-filter.md (hunk 2 above)
Comment (gdchamal) [inline _posts/rfc/2016-04-01-PI-filter.md]: OK 417 fit

File: _posts/rfc/2015-06-10-message-api.md (hunk 3 above)
Comment (gdchamal) [inline _posts/rfc/2015-06-10-message-api.md]: yes my bad, put sender back

File: _posts/rfc/2015-06-10-message-api.md [hunk 5]
```diff
@@ -29,37 +40,43 @@ Every **Thread** MUST provide the following information:
...
-  information be provided otherwise? It should remain consistent**
-* `sent_at`: date and time the message was sent (most probably extracted from
-  headers or metadata, depending on the message source)
-* `received_at`: date and time the message was received on the CaliOpen platform
+* `sender`: link to contact who sent this message.
```
Comment (iamdey) [inline _posts/rfc/2015-06-10-message-api.md]: the entity properties seems to be duplicated here and in recipients. I'm not sure about this for now

**Comment by iamdey** [comment]
anyway, LGTM

**Pull request merged**